seaborn/matrix.py

```diff
@@ -899,6 +913,10 @@ def color_list_to_matrix_and_cmap(colors, ind, axis=0):
             all_colors = set(itertools.chain(*colors))
             n = len(colors)
             m = len(colors[0])
+        elif isinstance(colors, pd.DataFrame):
```

Comment:
I'm not sure that it matters, but the order of priority in terms of checking for the different input types seems to vary across methods here. It would be good to try to be consistent.